### PR TITLE
DevKit updates for 4.x branch

### PR DESCRIPTION
**Pull request by SonataCI**

### CODE REVIEW

**Comment by jordisala1991** [comment]
I think the SF 7.0 compat is still missing a few changes.

File: src/ArgumentResolver/CompatibleValueResolverInterface.php [hunk 1]
```diff
@@ -23,7 +23,11 @@ interface CompatibleValueResolverInterface extends ValueResolverInterface
     {
     }
 } else {
-    /** @internal */
+    /**
+     * @internal
+     *
+     * @phpstan-ignore-next-line
```
Comment (jordisala1991) [inline src/ArgumentResolver/CompatibleValueResolverInterface.php]: This error can't be ignored like this.

**Comment by VincentLanglet** [comment]
> I think the SF 7.0 compat is still missing a few changes.

Build was green on https://github.com/sonata-project/SonataAdminBundle/pull/8127 (?)

**Comment by jordisala1991** [comment]
> > I think the SF 7.0 compat is still missing a few changes.
> 
> Build was green on #8127 (?)

Maybe between the RC and the stable version, something changed?

**Comment by VincentLanglet** [comment]
The check `0 !== preg_match("/7\..\../", Kernel::VERSION)` is not working on the `Test / PHP 8.1 + highest + normal` build because we're using Symfony 6.4 but phpunit-bridge 7.0

File: src/ArgumentResolver/CompatibleValueResolverInterface.php [hunk 2]
```diff
@@ -17,14 +17,14 @@
 use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
 
 // TODO: Remove this interface when dropping support of Symfony < 6.2 and replace its usage with ValueResolverInterface
-if (interface_exists(ValueResolverInterface::class)) {
+if (interface_exists(ArgumentValueResolverInterface::class)) {
     /** @internal */
-    interface CompatibleValueResolverInterface extends ValueResolverInterface
+    interface CompatibleValueResolverInterface extends ArgumentValueResolverInterface
```
Comment (Hanmac) [inline src/ArgumentResolver/CompatibleValueResolverInterface.php:22]: @VincentLanglet isn't it the other way around?

this code gives me deprecations right now

Comment (VincentLanglet) [reply src/ArgumentResolver/CompatibleValueResolverInterface.php:22]: Can you try
```
if (interface_exists(ArgumentValueResolverInterface::class) && !interface_exists(ValueResolverInterface::class)) {
    /** @internal */
    interface CompatibleValueResolverInterface extends ArgumentValueResolverInterface
    {
    }
} else {
    /** @internal */
    interface CompatibleValueResolverInterface extends ValueResolverInterface
    {
    }
}
```
?

If it works for you PR is welcome

Comment (Hanmac) [reply src/ArgumentResolver/CompatibleValueResolverInterface.php:22]: The Question is more about what to do between the versions:

* 3.1.0 - 6.2.0 there was only ArgumentValueResolverInterface
* 6.2.0 - 7.0.0 there are both ArgumentValueResolverInterface and ValueResolverInterface, the first one is deprecated.
* 7.0.0 - there is only ValueResolverInterface

what would be the right choice for 6.2.0 - 7.0.0 ? Which Interface should we extend?

there shouldn't be extra logic needed, its just ArgumentValueResolverInterface =>ValueResolverInterface


as it was before, using ValueResolverInterface if able was better?

Comment (VincentLanglet) [reply src/ArgumentResolver/CompatibleValueResolverInterface.php:22]: Feel free to open a pr with what you have in mind and try having a green build

Comment (Hanmac) [reply src/ArgumentResolver/CompatibleValueResolverInterface.php:22]: this should work:
https://github.com/sonata-project/SonataAdminBundle/pull/8139

**Pull request merged**